lib/pacemaker/pcmk_sched_allocate.c

```diff
@@ -2393,6 +2392,121 @@ order_first_probes(pe_working_set_t * data_set)
     }
 }
 
+static void
+order_first_probe_then_restart_repromote(pe_action_t * probe, pe_action_t * after)
+{
+    GListPtr gIter = NULL;
+
+    if (probe == NULL || safe_str_neq(probe->task, RSC_STATUS)) {
+        return;
+    }
+
+    if (after == NULL) {
+        return;
+    }
+
+    if (probe->rsc == NULL) {
+        return;
+    }
+
+    /* Avoid running into any possible loop */
+    if (is_set(after->flags, pe_action_tracking)) {
+        return;
+    }
+
+    pe_set_action_bit(after, pe_action_tracking);
+
+    crm_trace("Processing based on %s %s -> %s %s",
+              probe->uuid,
+              probe->node ? probe->node->details->uname: "",
+              after->uuid,
+              after->node ? after->node->details->uname : "");
+
+    if (after->rsc
+        && probe->rsc != after->rsc
+        && uber_parent(probe->rsc) != after->rsc) {
```

Comment:
There are pseudo clone actions between the instances . For example, an order chain could be like:  

A:0.probe ->A.start->B.start -> B:0.start

When tracing the order chain, there's "pe_order_implies_then" only between A.start->B.start. And there's only "pe_order_implies_first_printed | pe_order_runnable_left" between B.start -> B:0.start, which is kind of difficult to be relied on..the current approach processes it based on:
A:0.probe -> ... -> B.start

, adds an ordering like:
A:0.probe -> B.stop

Pseudo action B.start doesn't have a "node" though.

It's safe but indeed not optimal. Probably we shouldn't just rely on ordering in this case... I'll think of a better approach.

Reply:
Now it checks if there's any "compatible" child of an interleaved clone with find_compatible_child(),   proceeds through the ordering chain to the compatible/relevant child and builds the dependency with clone child rather than the clone itself.